packages/graph/src/graph2/graph.ts

```diff
@@ -818,6 +810,14 @@ export class Graph2 extends SVGZoomWidget {
         this._vertexG = this._renderElement.append("g");
 
         this._tooltip.target(domNode);
+
+        this.on("startMarqueeSelection", () => {
+        }).on("updateMarqueeSelection", rect => {
+            const vertices: VertexPlaceholder[] = this._graphData.vertices().filter(v => v.x >= rect.x && v.x <= rect.x + rect.width && v.y >= rect.y && v.y <= rect.y + rect.height);
```

Comment:
It's worth noting we could improve the performance of this selection using d3-quadtree.

Something like this:
```js
this.on("startMarqueeSelection", () => {
    this._quadtree = d3Quadtree(this._graphData.vertices(), v => v.x, v => v.y);
}).on("updateMarqueeSelection", rect => {
    const vertices = [];
    this._quadtree.visit((node, x1, y1, x2, y2) => {
        if (!node.length) {
            do {
                var d = node.data;
                if (d[0] >= rect.x && d[0] < rect.x + rect.width && d[1] >= rect.y && d[1] < rect.y + rect.height) {
                    vertices.push(d);
                }
            } while (node = node.next);
        }
        return x1 >= rect.x + rect.width || y1 >= rect.y + rect.height || x2 < rect.x + rect.width || y2 < rect.y;
    });
});
```

Reply:
Nice!